### PR TITLE
Fixed native framework evaluation

### LINKED ISSUE #5439: build\native\PackageName.targets isn't imported after installing package in C++ project
## Details about Problem

NuGet product used: VS UI or Package Manager Console
NuGet version (x.x.x.xxx): 4.3.0.4168
VS version (if appropriate): d15prerel 26615.05
OS version (i.e. win10 v1607 (14393.321)): rs2_release 15063.2.170323-1837

Worked before? If so, with which NuGet version: Yes in 4.2.0.2457 with d15svc 26430.13 (15.2)

## Detailed repro steps so we can see the same problem

1. Create new Blank C++ UWP project in VS
2. Install this package --> [XetNativeSdk.1.0.0.zip](https://github.com/NuGet/Home/files/1082172/XetNativeSdk.1.0.0.zip)
I tried two different approaches to installing it.  The bug repros either way.
a) In Manage NuGet Packages, added a custom package source pointing at its location on disk and then added package from Browse tab.
b) Added solution-level nuget.config containing packageSource pointing at its location on disk and adding package via Install-Package in the Package Manager Console.

**Expected:** XetNativeSdk is now visible under References in Solution Explorer.  vcxproj now contains an <Import> for XetNativeSdk.targets.

**Actual:** XetNativeSdk is not in References because .targets hasn't been imported.

### PR DESCRIPTION
**Pull request by alpaix**
Resolves NuGet/Home#5439.

Target framework for C++ projects should always be `native,v0.0`. The
inner logic of
`MSBuildProjectFrameworkUtility.GetProjectFrameworkStrings` is based on
project full path inspection. Given the project file extension is
`.vcxproj` utility method unconditionally returns `native` frameworks skipping
project system query. After LSL work native target framework evaluation
regressed due to wrong value for project path was provided.

This PR fixes the issue by using `FullName` instead of `FullPath` which
was a full path to a project directory. Also to avoid confusion in
future `FullPath` is renamed to `ProjectDirectory`. Dulplicate and
redundant utility methods are unified within `VsProjectAdapter.

//cc @debonte @rrelyea

### CODE REVIEW

**Comment by rohit21agrawal** [comment]
can we have some tests around this?

**Comment by alpaix** [comment]
Working on E2E tests now. Will send a separate commit.

File: src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectSystems/VsMSBuildProjectSystem.cs [hunk 1]
```diff
@@ -729,7 +729,7 @@ public VSLangProj.References References
             {
                 if (IsReferenceUnavailableException(e))
                 {
-                    var frameworkName = EnvDTEProjectInfoUtility.GetDotNetFrameworkName(VsProjectAdapter.Project);
+                    var frameworkName = await VsProjectAdapter.GetDotNetFrameworkNameAsync();
```
Comment (jainaashish) [inline src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/ProjectSystems/VsMSBuildProjectSystem.cs:732]: Should we handle the case where framework is null?

File: src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/VsProjectAdapter.cs [hunk 2]
```diff
@@ -340,11 +353,21 @@ public string Version
...
 
-            var projectPath = FullPath;
+            return NuGetFramework.UnsupportedFramework;
+        }
+
+        private async Task<string> GetTargetFrameworkStringAsync()
+        {
+            await _threadingService.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+            var projectPath = FullName;
```
Comment (jainaashish) [inline src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/VsProjectAdapter.cs:370]: 👍 I think then .jsproj would also have been broken...

File: test/EndToEnd/tests/UniversalWindowsTest.ps1 [hunk 3]
```diff
@@ -0,0 +1,25 @@
+function Test-UwpNativeAppInstallPackage {
+    [SkipTestForVS14()]
```
Comment (jainaashish) [inline test/EndToEnd/tests/UniversalWindowsTest.ps1]: any specific reason to skip these tests for dev14?

**Review by jainaashish — APPROVED** [review]
LGTM

**Pull request merged**